sov-modules/sov-modules-macros/src/dispatch/derive_rpc.rs

```diff
@@ -0,0 +1,285 @@
+use proc_macro2::Ident;
+use quote::{format_ident, quote, ToTokens};
+use syn::{Attribute, FnArg, ImplItem, Meta, MetaList, Path, PathSegment, Type, Signature, PatType};
+
+
+/// Retrns an attribute with the name `rpc_method` replaced with `method`, and the index
+/// into the argument array where the attribute was found.
+fn get_method_attribute(attributes: &[Attribute]) -> Option<(Attribute, usize)> {
+    for (idx, attribute) in attributes.iter().enumerate() {
+        if let Ok(Meta::List(MetaList { path, .. })) = attribute.parse_meta() {
+            if path.is_ident("rpc_method") {
+                let mut new_attr = attribute.clone();
+                let path = &mut new_attr.path;
+                path.segments.last_mut().unwrap().ident = format_ident!("method");
+                return Some((new_attr, idx));
+            }
+        }
+    }
+    None
+}
+
+fn intermediate_trait_name(ident: &Ident) -> Ident {
+    let mut ident_str = ident.to_string();
+    ident_str.push_str("Rpc");
+    format_ident!("{}", ident_str)
```

Comment:
I think I am missing something, but why is it not possible to do:

```
format_ident!({}Rpc", ident)
```
So there's less allocations

Reply:
You don't need this at all! Copilot generated it for me :). Removed the function in a later commit